### PR TITLE
Add data source for DX Router Configuration

### LINKED ISSUE #26432: New aws_dx_virtual_interface_router_configuration data source
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

In order to automate end 2 end the setup of a Direct Connect Connection using a VIF (Virtual Interface), it would be great to retrieve the sample router config using a data source terraform.

### New or Affected Resource(s)

Suggested name:

* aws_dx_virtual_interface_router_configuration

### Potential Terraform Configuration

```hcl
data "aws_dx_virtual_interface_router_configuration" "example" {
{
   "router_type_identifier": "string",
   "virtual_interface_id": "string"
}
```

**output example for CiscoSystemsInc-2900SeriesRouters-IOS124:**
```
! Amazon Web Services
!=======================================IPV4=======================================
! Direct Connect
! Virtual Interface ID: dxvif-fgh12345
!
! --------------------------------------------------------------------------------
! Interface Configuration

! interface port-channel 1 (In case of a LAG connection)

interface GigabitEthernet0/1
! channel-group 1 mode passive (In case of a LAG connection)
  no ip address

interface GigabitEthernet0/1.112
! interface port-channel 1.112 (In case of a LAG connection)
  description "Direct Connect to your Amazon VPC or AWS Cloud"
  encapsulation dot1Q 112
  ip address 169.254.96.6 255.255.255.248
...
```

### References

* https://docs.aws.amazon.com/directconnect/latest/APIReference/API_DescribeRouterConfiguration.html

### PR DESCRIPTION
**Pull request by schuylr**
### Description

Adds the `aws_dx_router_configuration` data source to allow users to download sample router configurations for their Direct Connect VIF.

### Relations

Closes #26432

### References

https://docs.aws.amazon.com/directconnect/latest/UserGuide/create-vif.html#vif-router-config 
https://docs.aws.amazon.com/sdk-for-go/api/service/directconnect/#DirectConnect.DescribeRouterConfiguration
https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/directconnect#Client.DescribeRouterConfiguration 
https://docs.aws.amazon.com/directconnect/latest/APIReference/API_DescribeRouterConfiguration.html 
https://docs.aws.amazon.com/directconnect/latest/APIReference/API_RouterType.html

### Output from Acceptance Testing

Note that a DX VIF must be provided via the `VIRTUAL_INTERFACE_ID` env variable.

```
terraform-provider-aws % AWS_DEFAULT_REGION=eu-west-1 VIRTUAL_INTERFACE_ID=dxvif-xxxxxxxx make testacc TESTS=TestAccDirectConnectRouterConfigurationDataSource_basic PKG=directconnect
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/directconnect/... -v -count 1 -parallel 20 -run='TestAccDirectConnectRouterConfigurationDataSource_basic'  -timeout 180m
=== RUN   TestAccDirectConnectRouterConfigurationDataSource_basic
=== PAUSE TestAccDirectConnectRouterConfigurationDataSource_basic
=== CONT  TestAccDirectConnectRouterConfigurationDataSource_basic
--- PASS: TestAccDirectConnectRouterConfigurationDataSource_basic (19.31s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/directconnect	23.515s
```

### CODE REVIEW

File: internal/service/directconnect/router_configuration_data_source.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+	}
+
+	return output, nil
+}
+
+func flattenRouter(apiObject *directconnect.RouterType) []interface{} {
+	tfMap := map[string]interface{}{}
+
+	if v := apiObject.Platform; v != nil {
+		tfMap["platform"] = v
```
Comment (GlennChia) [inline internal/service/directconnect/router_configuration_data_source.go:128]: We can use an in-built function to return the value of the string value that is passed 

```suggestion
		tfMap["platform"] = aws.StringValue(v)

File: internal/service/directconnect/router_configuration_data_source.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+	}
+
+	if v := apiObject.RouterTypeIdentifier; v != nil {
+		tfMap["router_type_identifier"] = v
```
Comment (GlennChia) [inline internal/service/directconnect/router_configuration_data_source.go:132]: We can use an in-built function to return the value of the string value that is passed

```suggestion
		tfMap["router_type_identifier"] = aws.StringValue(v)

File: internal/service/directconnect/router_configuration_data_source.go [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+	}
+
+	if v := apiObject.Software; v != nil {
+		tfMap["software"] = v
```
Comment (GlennChia) [inline internal/service/directconnect/router_configuration_data_source.go:136]: We can use an in-built function to return the value of the string value that is passed

```suggestion
		tfMap["software"] = aws.StringValue(v)

File: internal/service/directconnect/router_configuration_data_source.go [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+	}
+
+	if v := apiObject.Vendor; v != nil {
+		tfMap["vendor"] = v
```
Comment (GlennChia) [inline internal/service/directconnect/router_configuration_data_source.go:140]: We can use an in-built function to return the value of the string value that is passed

```suggestion
		tfMap["vendor"] = aws.StringValue(v)

File: internal/service/directconnect/router_configuration_data_source.go [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+	}
+
+	if v := apiObject.XsltTemplateName; v != nil {
+		tfMap["xslt_template_name"] = v
```
Comment (GlennChia) [inline internal/service/directconnect/router_configuration_data_source.go:144]: We can use an in-built function to return the value of the string value that is passed

```suggestion
		tfMap["xslt_template_name"] = aws.StringValue(v)

File: internal/service/directconnect/router_configuration_data_source.go [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+	}
+
+	if v := apiObject.XsltTemplateNameForMacSec; v != nil {
+		tfMap["xslt_template_name_for_mac_sec"] = v
```
Comment (GlennChia) [inline internal/service/directconnect/router_configuration_data_source.go:148]: We can use an in-built function to return the value of the string value that is passed

```suggestion
		tfMap["xslt_template_name_for_mac_sec"] = aws.StringValue(v)

File: internal/service/directconnect/router_configuration_data_source_test.go [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+					resource.TestCheckResourceAttr(dataSourceName, "router.0.vendor", "Cisco Systems, Inc."),
+					resource.TestCheckResourceAttr(dataSourceName, "router.0.xslt_template_name", "customer-router-cisco-generic.xslt"),
+					resource.TestCheckResourceAttr(dataSourceName, "router.0.xslt_template_name_for_mac_sec", ""),
+				),
+			},
+		},
+	})
+}
+
+func testAccRouterConfigurationDataSourceConfig_basic(virtualInterfaceId string, routerTypeIdentifier string) string {
```
Comment (GlennChia) [inline internal/service/directconnect/router_configuration_data_source_test.go:49]: Small styling change.

```suggestion
func testAccRouterConfigurationDataSourceConfig_basic(virtualInterfaceId, routerTypeIdentifier string) string {

File: website/docs/d/dx_router_configuration.html.markdown [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+
+## Example Usage
+
+### Basic Usage
+
+```terraform
+data "aws_dx_router_configuration" "example" {
+  virtual_interface_id = "dxvif-abcde123"
+  router_type_identifier = "CiscoSystemsInc-2900SeriesRouters-IOS124"
+}
```
Comment (GlennChia) [inline website/docs/d/dx_router_configuration.html.markdown:21]: styling change `terraform fmt`

```suggestion
data "aws_dx_router_configuration" "example" {
  virtual_interface_id   = "dxvif-abcde123"
  router_type_identifier = "CiscoSystemsInc-2900SeriesRouters-IOS124"
}

**Review by GlennChia — CHANGES_REQUESTED** [review]
PR looks good, requesting small styling changes, no functional changes required.

**Review by GlennChia — APPROVED** [review]
LGTM!

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTS=TestAccDirectConnectRouterConfigurationDataSource_ PKG=directconnect
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/directconnect/... -v -count 1 -parallel 20 -run='TestAccDirectConnectRouterConfigurationDataSource_'  -timeout 180m
=== RUN   TestAccDirectConnectRouterConfigurationDataSource_basic
    router_configuration_data_source_test.go:17: Environment variable VIRTUAL_INTERFACE_ID is not set
--- SKIP: TestAccDirectConnectRouterConfigurationDataSource_basic (0.00s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/directconnect	4.891s
```

**Comment by ewbankkit** [comment]
@schuylr Thanks for the contribution :tada: :clap:.

**Pull request merged**